core/src/main/java/org/infinispan/interceptors/impl/BaseStateTransferInterceptor.java

```diff
@@ -54,21 +55,28 @@
 
    @Inject Configuration configuration;
    @Inject protected StateTransferLock stateTransferLock;
-   @Inject @ComponentName(KnownComponentNames.NON_BLOCKING_EXECUTOR)
+   @Inject
+   @ComponentName(KnownComponentNames.NON_BLOCKING_EXECUTOR)
    Executor nonBlockingExecutor;
    @Inject DistributionManager distributionManager;
-   @Inject @ComponentName(KnownComponentNames.TIMEOUT_SCHEDULE_EXECUTOR)
+   @Inject
+   @ComponentName(KnownComponentNames.TIMEOUT_SCHEDULE_EXECUTOR)
    ScheduledExecutorService timeoutExecutor;
 
    private long transactionDataTimeout;
    private boolean isScattered;
 
-   private final InvocationFinallyFunction<GetKeysInGroupCommand> handleLocalGetKeysInGroupReturn = this::handleLocalGetKeysInGroupReturn;
+   private final InvocationFinallyFunction<GetKeysInGroupCommand> handleLocalGetKeysInGroupReturn =
+         this::handleLocalGetKeysInGroupReturn;
 
    @Start
    public void start() {
-      transactionDataTimeout = configuration.clustering().remoteTimeout();
       isScattered = configuration.clustering().cacheMode().isScattered();
+      transactionDataTimeout = configuration.clustering().remoteTimeout();
+      configuration.clustering().attributes().attribute(ClusteringConfiguration.REMOTE_TIMEOUT)
+                   .addListener((a, ignored) -> {
+                      transactionDataTimeout = a.get();
+                   });
```

Comment:
I guess you accidently hit formatting for this file. It would be better if we could just include the timeout additions here.